src/pubsub/ua_pubsub_reader.c

```diff
@@ -218,9 +218,10 @@ UA_DataSetReader_generateDataSetMessage(UA_Server *server, UA_DataSetMessage *da
 }
 
 static UA_StatusCode
-UA_DataSetReader_generateNetworkMessage(UA_PubSubConnection *pubSubConnection, UA_DataSetReader *dataSetReader,
+UA_DataSetReader_generateNetworkMessage(UA_PubSubConnection *pubSubConnection, UA_ReaderGroup *readerGroup,
                                         UA_DataSetMessage *dsm, UA_UInt16 *writerId, UA_Byte dsmCount,
                                         UA_NetworkMessage *nm) {
+    UA_DataSetReader *dataSetReader = LIST_FIRST(&readerGroup->readers);
```

Comment:
Always taking the first reader prevents us generating a network message for the other readers when there is more than one.

Better forward the reader **and** the readergroup.